src/main/java/org/folio/services/locationunit/InstitutionService.java

```diff
@@ -0,0 +1,108 @@
+import org.folio.rest.jaxrs.resource.LocationUnits.DeleteLocationUnitsInstitutionsByIdResponse;
+import org.folio.rest.jaxrs.resource.LocationUnits.DeleteLocationUnitsInstitutionsResponse;
+import org.folio.rest.jaxrs.resource.LocationUnits.GetLocationUnitsInstitutionsByIdResponse;
+import org.folio.rest.jaxrs.resource.LocationUnits.GetLocationUnitsInstitutionsResponse;
+import org.folio.rest.jaxrs.resource.LocationUnits.PostLocationUnitsInstitutionsResponse;
+import org.folio.rest.jaxrs.resource.LocationUnits.PutLocationUnitsInstitutionsByIdResponse;
+import org.folio.rest.persist.PgUtil;
+import org.folio.services.domainevent.InstitutionDomainEventPublisher;
+
+@Log4j2
+public class InstitutionService {
+
+  public static final String INSTITUTION_TABLE = "locinstitution";
+  private static final String MSG_ID_NOT_MATCH =
+    "Illegal operation: Institution ID cannot be changed";
+
+  private final Context vertxContext;
+  private final Map<String, String> okapiHeaders;
+  private final InstitutionRepository repository;
+  private final InstitutionDomainEventPublisher domainEventService;
+
+  public InstitutionService(Context vertxContext,
+                            Map<String, String> okapiHeaders) {
+    this.vertxContext = vertxContext;
+    this.okapiHeaders = okapiHeaders;
+    this.repository = new InstitutionRepository(vertxContext, okapiHeaders);
+    this.domainEventService =
+      new InstitutionDomainEventPublisher(vertxContext, okapiHeaders);
+  }
+
+  public Future<Response> getByQuery(String cql, int offset, int limit) {
+    return PgUtil.get(INSTITUTION_TABLE, Locinst.class, Locinsts.class, cql,
+      offset, limit, okapiHeaders, vertxContext,
+      GetLocationUnitsInstitutionsResponse.class);
+  }
+
+  public Future<Response> getById(String id) {
+    return PgUtil.getById(INSTITUTION_TABLE, Locinst.class, id, okapiHeaders,
+      vertxContext, GetLocationUnitsInstitutionsByIdResponse.class);
+  }
+
+  public Future<Response> create(Locinst institution) {
+    return PgUtil.post(INSTITUTION_TABLE, institution, okapiHeaders,
+        vertxContext, PostLocationUnitsInstitutionsResponse.class)
+      .onSuccess(response ->  domainEventService.publishCreated());
+  }
+
+  public Future<Response> update(String id, Locinst institution) {
+    if (Objects.nonNull(institution.getId()) && !institution.getId().equals(id)) {
+      return succeededFuture(
+        PutLocationUnitsInstitutionsByIdResponse.respond400WithTextPlain(
+          createValidationErrorMessage("locinst", institution.getId(),
+            MSG_ID_NOT_MATCH)));
+    }
+
+    if (Objects.isNull(institution.getId())) {
+      institution.setId(id);
+    }
+
+    return repository.getById(id)
+      .compose(oldInstitution ->
+        PgUtil.put(INSTITUTION_TABLE, institution, id,
+          okapiHeaders, vertxContext,
+          PutLocationUnitsInstitutionsByIdResponse.class)
+        .onSuccess(domainEventService.publishUpdated(institution)));
```

Comment:
please check this place in action. How it looks. We shuold have first oldInstitution and the next new instution in kafka message. 
example
![image](https://github.com/folio-org/mod-inventory-storage/assets/113523904/098a24c5-dd56-409c-a80f-9ffaf686524d)

Reply:
> I suppose instutition should be oldInstutition. please check this place in action. How it looks. We shuold have first oldInstitution and the next new instution in kafka message. example ![image](https://private-user-images.githubusercontent.com/113523904/336820322-098a24c5-dd56-409c-a80f-9ffaf686524d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTc1OTEyMjAsIm5iZiI6MTcxNzU5MDkyMCwicGF0aCI6Ii8xMTM1MjM5MDQvMzM2ODIwMzIyLTA5OGEyNGM1LWRkNTYtNDA5Yy1hODBmLTlmZmFmNjg2NTI0ZC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNjA1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDYwNVQxMjM1MjBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1kZmE1Njk5YTc0Njk3YTg4NTk1MzJhZDI3YjdlOTRjNTUzZmFiNWRkNmJkMDkwYjcyZTk1N2FkOTRmNDkxMGY5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.ZYHlB-uYEUQbwHR_1z3Bz1JNZY_PwxlfGHxXzJ51D28)

Done, fixed my update code, here are topic messages on update:

![image](https://github.com/folio-org/mod-inventory-storage/assets/169707997/e1c7f82f-7a48-4fa2-8809-75ff24c3ba9e)